wnfs-wasm/src/fs/utils.rs

```diff
@@ -67,11 +77,20 @@ pub(crate) fn create_private_op_result<T: Into<JsValue>>(
     Ok(value!(op_result))
 }
 
-pub(crate) fn error<E>(message: &str) -> impl FnOnce(E) -> js_sys::Error + '_
-where
-    E: Debug,
-{
-    move |e| Error::new(&format!("{message}: {e:?}"))
+pub(crate) fn create_private_file_result(
+    file: WnfsPrivateFile,
+    hamt: Rc<WnfsPrivateForest>,
+) -> JsResult<JsValue> {
+    let op_result = Object::new();
+
+    Reflect::set(
+        &op_result,
+        &value!("file"),
+        &PrivateFile(Rc::new(file)).into(),
+    )?;
+    Reflect::set(&op_result, &value!("hamt"), &PrivateForest(hamt).into())?;
```

Comment:
dumb q: why not return the hamt directly, instead of keyed to "hamt" in an JS object? Feels strange to label the "hamt" like this, but I'm probably missing something.

Reply:
BTW, we can't simply return an hamt. We need to somehow convert it to a JsValue and Reflect.set is the only way I know of to do that with wasm-bindgen. On that note, I can change it to an Array and still Reflect.set on it. We should probably do that the `ls` operation too later.

Reply:
right, I knew not the direct hamt itself, but the object w/ key "hamt" felt weird.